diff --git a/src/Asset/CssCollectionOptimizer.php b/src/Asset/CssCollectionOptimizer.php
index 0f2296a..4ab1679 100644
--- a/src/Asset/CssCollectionOptimizer.php
+++ b/src/Asset/CssCollectionOptimizer.php
@@ -28,7 +28,7 @@ class CssCollectionOptimizer extends DrupalCssCollectionOptimizer {
           $file_system->delete($uri);
         }
         catch (\Exception $e) {
-          $this->getLogger('flysystem')->error($e->getMessage());
+          \Drupal::service('logger.factory')->get('flysystem')->error($e->getMessage());
         }
       }
     };
diff --git a/src/Asset/JsCollectionOptimizer.php b/src/Asset/JsCollectionOptimizer.php
index 6216288..ca61254 100644
--- a/src/Asset/JsCollectionOptimizer.php
+++ b/src/Asset/JsCollectionOptimizer.php
@@ -28,7 +28,7 @@ class JsCollectionOptimizer extends DrupalJsCollectionOptimizer {
           $file_system->delete($uri);
         }
         catch (\Exception $e) {
-          $this->getLogger('flysystem')->error($e->getMessage());
+          \Drupal::service('logger.factory')->get('flysystem')->error($e->getMessage());
         }
       }
     };
