diff --git a/src/EventSubscriber/EnsureSubscriber.php b/src/EventSubscriber/EnsureSubscriber.php
index 27f3795..d3ffec1 100644
--- a/src/EventSubscriber/EnsureSubscriber.php
+++ b/src/EventSubscriber/EnsureSubscriber.php
@@ -6,7 +6,7 @@ use Drupal\Core\Logger\RfcLogLevel;
 use Drupal\flysystem\Event\EnsureEvent;
 use Drupal\flysystem\Event\FlysystemEvents;
 use Psr\Log\LoggerInterface;
-use Symfony\Contract\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
 /**
