diff --git a/tests/src/Unit/Flysystem/MissingTest.php b/tests/src/Unit/Flysystem/MissingTest.php
index d69a8bb..0421e4c 100644
--- a/tests/src/Unit/Flysystem/MissingTest.php
+++ b/tests/src/Unit/Flysystem/MissingTest.php
@@ -16,7 +16,7 @@ class MissingTest extends UnitTestCase {
    * @covers \Drupal\flysystem\Flysystem\Missing
    */
   public function test() {
-    $plugin = new Missing([]);
+    $plugin = new Missing();
     $this->assertInstanceOf(MissingAdapter::class, $plugin->getAdapter());
     $this->assertTrue(is_array($plugin->ensure()));
     $this->assertCount(1, $plugin->ensure());
