diff --git a/src/Asset/CssCollectionOptimizerLazy.php b/src/Asset/CssCollectionOptimizerLazy.php
deleted file mode 100644
index c47de1d..0000000
--- a/src/Asset/CssCollectionOptimizerLazy.php
+++ /dev/null
@@ -1,27 +0,0 @@
-<?php
-
-namespace Drupal\flysystem\Asset;
-
-use Drupal\Core\Asset\CssCollectionOptimizerLazy as DrupalCssCollectionOptimizerLazy;
-use Drupal\Core\File\Exception\FileException;
-
-/**
- * Optimizes CSS assets.
- */
-class CssCollectionOptimizerLazy extends DrupalCssCollectionOptimizerLazy {
-
-  use SchemeExtensionTrait;
-
-  /**
-   * {@inheritdoc}
-   */
-  public function deleteAll() {
-    try {
-      $this->fileSystem->deleteRecursive($this->getSchemeForExtension('css') . '://css');
-    }
-    catch (FileException $fileException) {
-      \Drupal::logger('flysystem')->error($fileException->getMessage());// @phpstan-ignore-line @codingStandardsIgnoreLine 
-    }
-  }
-
-}
diff --git a/src/Asset/JsCollectionOptimizer.php b/src/Asset/JsCollectionOptimizer.php
index 3a3b26b..6238c81 100644
--- a/src/Asset/JsCollectionOptimizer.php
+++ b/src/Asset/JsCollectionOptimizer.php
@@ -3,7 +3,6 @@
 namespace Drupal\flysystem\Asset;
 
 use Drupal\Core\Asset\JsCollectionOptimizer as DrupalJsCollectionOptimizer;
-use Drupal\Core\Logger\LoggerChannelTrait;
 
 /**
  * Optimizes JavaScript assets.
@@ -11,7 +10,6 @@ use Drupal\Core\Logger\LoggerChannelTrait;
 class JsCollectionOptimizer extends DrupalJsCollectionOptimizer {// @phpstan-ignore-line @codingStandardsIgnoreLine 
 
   use SchemeExtensionTrait;
-  use LoggerChannelTrait;
 
   /**
    * {@inheritdoc}
diff --git a/src/Asset/JsCollectionOptimizerLazy.php b/src/Asset/JsCollectionOptimizerLazy.php
deleted file mode 100644
index c7f29af..0000000
--- a/src/Asset/JsCollectionOptimizerLazy.php
+++ /dev/null
@@ -1,27 +0,0 @@
-<?php
-
-namespace Drupal\flysystem\Asset;
-
-use Drupal\Core\Asset\JsCollectionOptimizerLazy as DrupalJsCollectionOptimizerLazy;
-use Drupal\Core\File\Exception\FileException;
-
-/**
- * Optimizes JavaScript assets.
- */
-class JsCollectionOptimizerLazy extends DrupalJsCollectionOptimizerLazy {// @phpstan-ignore-line @codingStandardsIgnoreLine 
-
-  use SchemeExtensionTrait;
-
-  /**
-   * {@inheritdoc}
-   */
-  public function deleteAll() {
-    try {
-      $this->fileSystem->deleteRecursive($this->getSchemeForExtension('js') . '://js');
-    }
-    catch (FileException $fileException) {
-      \Drupal::logger('flysystem')->error($fileException->getMessage());// @phpstan-ignore-line @codingStandardsIgnoreLine 
-    }
-  }
-
-}
diff --git a/src/Asset/SchemeExtensionTrait.php b/src/Asset/SchemeExtensionTrait.php
index 9fb2b30..4a58c41 100644
--- a/src/Asset/SchemeExtensionTrait.php
+++ b/src/Asset/SchemeExtensionTrait.php
@@ -21,12 +21,12 @@ trait SchemeExtensionTrait {
    */
   public function getSchemeForExtension($extension) {
     $has_assets_scheme = class_exists(AssetsStream::class);
-    $extension_scheme = 'public';
+    $extension_scheme = 'assets';
 
     foreach (Settings::get('flysystem', []) as $scheme => $configuration) {
       if (!empty($configuration['serve_' . $extension]) && !empty($configuration['driver'])) {
         if ($has_assets_scheme) {
-          @trigger_error(sprintf('The serve_%s Flysystem option is deprecated in flysystem:2.1.0 and is removed from flysystem:3.0.0. Use the assets:// stream wrapper instead. See https://www.drupal.org/node/3328126', $extension), E_USER_DEPRECATED);
+          @trigger_error(sprintf('The serve_%s Flysystem option is deprecated in flysystem:2.1.0 and is removed from flysystem:2.3.0. Use the assets:// stream wrapper instead. See https://www.drupal.org/node/3328126', $extension), E_USER_DEPRECATED);
         }
         // Don't break, the last configured one will win.
         $extension_scheme = $scheme;
diff --git a/src/FlysystemServiceProvider.php b/src/FlysystemServiceProvider.php
index 1c2590d..d245320 100644
--- a/src/FlysystemServiceProvider.php
+++ b/src/FlysystemServiceProvider.php
@@ -65,22 +65,20 @@ class FlysystemServiceProvider implements ServiceProviderInterface {
       return;
     }
 
-    $container
-      ->getDefinition('asset.' . $extension . '.dumper')
-      ->setClass('Drupal\flysystem\Asset\AssetDumper');
-
     $optimizer = $container->getDefinition('asset.' . $extension . '.collection_optimizer');
     if ($optimizer->getClass() === 'Drupal\Core\Asset\\' . ucfirst($extension) . 'CollectionOptimizer') {
+      @trigger_error(sprintf('The serve_%s Flysystem option is deprecated in flysystem:2.1.0 and is removed from flysystem:2.3.0. Use the assets:// stream wrapper instead. See https://www.drupal.org/node/3328126', $extension), E_USER_DEPRECATED);
       $optimizer->setClass('Drupal\flysystem\Asset\\' . ucfirst($extension) . 'CollectionOptimizer');
     }
-    if ($optimizer->getClass() === 'Drupal\Core\Asset\\' . ucfirst($extension) . 'CollectionOptimizerLazy') {
-      $optimizer->setClass('Drupal\flysystem\Asset\\' . ucfirst($extension) . 'CollectionOptimizerLazy');
-    }
+
+    $container
+      ->getDefinition('asset.' . $extension . '.dumper')
+      ->setClass('Drupal\flysystem\Asset\AssetDumper');
 
     if ($extension === 'css') {
       $container
-        ->getDefinition('asset.css.optimizer')
-        ->setClass('Drupal\flysystem\Asset\CssOptimizer');
+        ->getDefinition('asset.' . $extension . '.optimizer')
+        ->setClass('Drupal\flysystem\Asset\\' . ucfirst($extension) . 'Optimizer');
     }
   }
 
diff --git a/tests/src/Kernel/CollectionOptimizerTest.php b/tests/src/Kernel/CollectionOptimizerTest.php
index 04c7877..bf84240 100644
--- a/tests/src/Kernel/CollectionOptimizerTest.php
+++ b/tests/src/Kernel/CollectionOptimizerTest.php
@@ -4,7 +4,6 @@ namespace Drupal\Tests\flysystem\Kernel;
 
 use Drupal\Core\Asset\AssetCollectionGrouperInterface;
 use Drupal\Core\Asset\AssetOptimizerInterface;
-use Drupal\Core\Asset\CssOptimizer;
 use Drupal\Core\State\StateInterface;
 use Drupal\flysystem\Asset\AssetDumper;
 use Drupal\flysystem\Asset\CssCollectionOptimizer;
@@ -122,8 +121,9 @@ class CollectionOptimizerTest extends KernelTestBase {
     $grouper = $this->prophesize(AssetCollectionGrouperInterface::class);
     $dumper = $this->prophesize(AssetDumper::class);
     $state = $this->prophesize(StateInterface::class);
+    $cs_optimizer = $this->prophesize(AssetOptimizerInterface::class);
 
-    $optimizer = new CssCollectionOptimizer($grouper->reveal(), new CssOptimizer($this->fileUrlGenerator), $dumper->reveal(), $state->reveal(), $this->container->get('file_system'));
+    $optimizer = new CssCollectionOptimizer($grouper->reveal(), $cs_optimizer->reveal(), $dumper->reveal(), $state->reveal(), $this->container->get('file_system'));
     $optimizer->deleteAll();
 
     foreach ($this->cssFilesUnderTest() as $css_file => $expired) {
diff --git a/tests/src/Unit/Asset/SchemeExtensionTraitTest.php b/tests/src/Unit/Asset/SchemeExtensionTraitTest.php
index a9a01a9..dd4fdce 100644
--- a/tests/src/Unit/Asset/SchemeExtensionTraitTest.php
+++ b/tests/src/Unit/Asset/SchemeExtensionTraitTest.php
@@ -26,8 +26,8 @@ class SchemeExtensionTraitTest extends UnitTestCase {
 
     $trait = $this->getMockForTrait(SchemeExtensionTrait::class);
     $this->assertSame('local', $trait->getSchemeForExtension('js'));
-    $this->assertSame('public', $trait->getSchemeForExtension('css'));
-    $this->assertSame('public', $trait->getSchemeForExtension('jpg'));
+    $this->assertSame('assets', $trait->getSchemeForExtension('css'));
+    $this->assertSame('assets', $trait->getSchemeForExtension('jpg'));
   }
 
 }
