diff --git a/src/FlysystemBridge.php b/src/FlysystemBridge.php
index 1f4dbce..a0e64a1 100644
--- a/src/FlysystemBridge.php
+++ b/src/FlysystemBridge.php
@@ -4,6 +4,7 @@ namespace Drupal\flysystem;
 
 use Drupal\Core\StreamWrapper\StreamWrapperInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
+use League\Flysystem\FilesystemInterface;
 use League\Flysystem\Util;
 use Twistor\FlysystemStreamWrapper;
 
@@ -117,7 +118,7 @@ class FlysystemBridge extends FlysystemStreamWrapper implements StreamWrapperInt
   /**
    * {@inheritdoc}
    */
-  protected function getFilesystem() {
+  protected function getFilesystem(): FilesystemInterface {
     if (!isset($this->filesystem)) {
       $this->filesystem = $this->getFilesystemForScheme($this->getProtocol());
     }
