diff --git a/tests/src/Kernel/CollectionOptimizerTest.php b/tests/src/Kernel/CollectionOptimizerTest.php
index 9fb14f7..043d3ac 100644
--- a/tests/src/Kernel/CollectionOptimizerTest.php
+++ b/tests/src/Kernel/CollectionOptimizerTest.php
@@ -4,6 +4,8 @@ namespace Drupal\Tests\flysystem\Kernel;
 
 use Drupal\Core\Asset\AssetCollectionGrouperInterface;
 use Drupal\Core\Asset\CssOptimizer;
+use Drupal\Core\Asset\CssCollectionGrouper;
+use Drupal\Core\Asset\JsCollectionGrouper;
 use Drupal\Core\Asset\JsOptimizer;
 use Drupal\Core\State\StateInterface;
 use Drupal\flysystem\Asset\AssetDumper;
@@ -32,12 +34,35 @@ class CollectionOptimizerTest extends KernelTestBase {
    */
   protected $fileUrlGenerator;
 
+  /**
+   * Flysystem Logger channel.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $loggerChannel;
+
+  /**
+   * JsOptimizer.
+   *
+   * @var \Drupal\Core\Asset\JsOptimizer
+   */
+  protected $optimizer;
+
+  /**
+   * AssetCollectionGrouperInterface.
+   *
+   * @var \Drupal\Core\Asset\AssetCollectionGrouperInterface|\Drupal\Core\Asset\CssCollectionGrouper|\Drupal\Core\Asset\JsCollectionGrouper
+   */
+  protected $grouper;
+
   /**
    * {@inheritdoc}
    */
   public function setUp(): void {
     parent::setUp();
     $this->fileUrlGenerator = $this->container->get('file_url_generator');
+    $logger = $this->createMock('\Psr\Log\LoggerInterface');
+    $this->optimizer = new JsOptimizer($logger);
     $this->cleanUp();
   }
 
@@ -72,11 +97,11 @@ class CollectionOptimizerTest extends KernelTestBase {
       touch($js_file, \Drupal::time()->getRequestTime() - 2591999);
     }
 
-    $grouper = $this->prophesize(AssetCollectionGrouperInterface::class);
     $dumper = $this->prophesize(AssetDumper::class);
     $state = $this->prophesize(StateInterface::class);
+    $this->grouper = new CssCollectionGrouper();
 
-    $optimizer = new JsCollectionOptimizer($grouper->reveal(), new JsOptimizer(), $dumper->reveal(), $state->reveal(), $this->container->get('file_system'));
+    $optimizer = new JsCollectionOptimizer($this->grouper, $this->optimizer, $dumper->reveal(), $state->reveal(), $this->container->get('file_system'));
     $optimizer->deleteAll();
 
     foreach ($this->jsFilesUnderTest() as $js_file => $expired) {
