diff --git a/src/FlysystemStreamWrapper.php b/src/FlysystemStreamWrapper.php
index c0ae3bc..5ef110c 100755
--- a/src/FlysystemStreamWrapper.php
+++ b/src/FlysystemStreamWrapper.php
@@ -14,7 +14,7 @@ use League\Flysystem\Visibility;
 use M2MTech\FlysystemStreamWrapper\Flysystem\Helper\UserGuesser;
 use M2MTech\FlysystemStreamWrapper\Flysystem\StreamWrapper;
 
-final class FlysystemStreamWrapper
+class FlysystemStreamWrapper
 {
     public const LOCK_STORE = 'lock_store';
     public const LOCK_TTL = 'lock_ttl';
