diff --git a/src/Plugin/FlysystemUrlTrait.php b/src/Plugin/FlysystemUrlTrait.php
index 17568dd..cd27f7c 100644
--- a/src/Plugin/FlysystemUrlTrait.php
+++ b/src/Plugin/FlysystemUrlTrait.php
@@ -25,9 +25,9 @@ trait FlysystemUrlTrait {
    */
   public function getExternalUrl($uri) {
     $path = str_replace('\\', '/', $this->getTarget($uri));
-
+    $scheme = $this->getScheme($uri);
     $arguments = [
-      'scheme' => $this->getScheme($uri),
+      'scheme' => $scheme,
       'filepath' => $path,
     ];
     return Url::fromRoute('flysystem.serve', $arguments, ['absolute' => TRUE])->toString();
@@ -41,38 +41,11 @@ trait FlysystemUrlTrait {
    *
    * @return string
    *   The file path of the URI.
-   */
-  protected function getTarget($uri) {
-    return $this->normalizeRelativePath(substr($uri, strpos($uri, '://') + 3));
-  }
-
-  /**
-   * Returns the scheme from the internal URI.
-   *
-   * @param string $uri
-   *   The URI.
-   *
-   * @return string
-   *   The scheme.
-   */
-  protected function getScheme($uri) {
-    return substr($uri, 0, strpos($uri, '://'));
-  }
-
-  /**
-   * Normalize relative directories in a path.
-   *
-   * @param string $path
-   *   Path containing directories to normalize.
-   *
-   * @return string
-   *   Normalized path.
    *
    * @throws \LogicException
-   *
-   * @see FlysystemStreamWrapper::normalizeRelativePath()
    */
-  protected function normalizeRelativePath($path) {
+  protected function getTarget($uri) {
+    $path = substr($uri, strpos($uri, '://') + 3);
     $path = str_replace('\\', '/', $path);
 
     while (preg_match('#\p{C}+|^\./#u', $path)) {
@@ -103,6 +76,20 @@ trait FlysystemUrlTrait {
     }
 
     return implode('/', $parts);
+
+  }
+
+  /**
+   * Returns the scheme from the internal URI.
+   *
+   * @param string $uri
+   *   The URI.
+   *
+   * @return string
+   *   The scheme.
+   */
+  protected function getScheme($uri) {
+    return substr($uri, 0, strpos($uri, '://'));
   }
 
 }
diff --git a/src/StreamWrapper/Exception/DirectoryExistsException.php b/src/StreamWrapper/Exception/DirectoryExistsException.php
index c4d29c4..9523fd4 100644
--- a/src/StreamWrapper/Exception/DirectoryExistsException.php
+++ b/src/StreamWrapper/Exception/DirectoryExistsException.php
@@ -4,6 +4,8 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * Directory Exists exception.
+ *
+ * @codeCoverageIgnore
  */
 class DirectoryExistsException extends TriggerErrorException {
 
diff --git a/src/StreamWrapper/Exception/DirectoryNotEmptyException.php b/src/StreamWrapper/Exception/DirectoryNotEmptyException.php
index d040ddb..a56a1d2 100644
--- a/src/StreamWrapper/Exception/DirectoryNotEmptyException.php
+++ b/src/StreamWrapper/Exception/DirectoryNotEmptyException.php
@@ -4,6 +4,8 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * Directory Not Empty Exception.
+ *
+ * @codeCoverageIgnore
  */
 class DirectoryNotEmptyException extends TriggerErrorException {
 
diff --git a/src/StreamWrapper/Exception/Exception.php b/src/StreamWrapper/Exception/Exception.php
index 866eb96..fcde5fd 100644
--- a/src/StreamWrapper/Exception/Exception.php
+++ b/src/StreamWrapper/Exception/Exception.php
@@ -4,5 +4,7 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * General exception triggered when no specified exception catches issue.
+ *
+ * @codeCoverageIgnore
  */
 class Exception extends \Exception {}
diff --git a/src/StreamWrapper/Exception/FileNotFoundException.php b/src/StreamWrapper/Exception/FileNotFoundException.php
index dd5126e..d40e77b 100644
--- a/src/StreamWrapper/Exception/FileNotFoundException.php
+++ b/src/StreamWrapper/Exception/FileNotFoundException.php
@@ -6,6 +6,8 @@ use Drupal\flysystem\StreamWrapper\Exception\Exception as BaseException;
 
 /**
  * File Not Found exception.
+ *
+ * @codeCoverageIgnore
  */
 class FileNotFoundException extends Exception {
   /**
diff --git a/src/StreamWrapper/Exception/NotADirectoryException.php b/src/StreamWrapper/Exception/NotADirectoryException.php
index c542c7e..8331178 100644
--- a/src/StreamWrapper/Exception/NotADirectoryException.php
+++ b/src/StreamWrapper/Exception/NotADirectoryException.php
@@ -4,6 +4,8 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * Not a Directory Exception.
+ *
+ * @codeCoverageIgnore
  */
 class NotADirectoryException extends TriggerErrorException {
 
diff --git a/src/StreamWrapper/Exception/RootViolationException.php b/src/StreamWrapper/Exception/RootViolationException.php
index 7aa626a..87e50a5 100644
--- a/src/StreamWrapper/Exception/RootViolationException.php
+++ b/src/StreamWrapper/Exception/RootViolationException.php
@@ -4,6 +4,8 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * Root Violation.
+ *
+ * @codeCoverageIgnore
  */
 class RootViolationException extends \LogicException {
 }
diff --git a/src/StreamWrapper/Exception/TriggerErrorException.php b/src/StreamWrapper/Exception/TriggerErrorException.php
index eb560cc..01c60ff 100644
--- a/src/StreamWrapper/Exception/TriggerErrorException.php
+++ b/src/StreamWrapper/Exception/TriggerErrorException.php
@@ -4,6 +4,8 @@ namespace Drupal\flysystem\StreamWrapper\Exception;
 
 /**
  * Trigger Error Exception.
+ *
+ * @codeCoverageIgnore
  */
 class TriggerErrorException extends Exception {
   /**
diff --git a/src/StreamWrapper/FlysystemStreamWrapper.php b/src/StreamWrapper/FlysystemStreamWrapper.php
index 2647601..79d1bea 100644
--- a/src/StreamWrapper/FlysystemStreamWrapper.php
+++ b/src/StreamWrapper/FlysystemStreamWrapper.php
@@ -15,6 +15,8 @@ use League\Flysystem\FilesystemInterface;
 
 /**
  * An adapter for Flysystem to a PHP stream wrapper.
+ *
+ * @codeCoverageIgnore
  */
 class FlysystemStreamWrapper {
   /**
@@ -154,84 +156,6 @@ class FlysystemStreamWrapper {
    */
   protected $uri;
 
-  /**
-   * Registers the stream wrapper protocol if not already registered.
-   *
-   * @param string $protocol
-   *   The protocol.
-   * @param \League\Flysystem\FilesystemInterface $filesystem
-   *   The filesystem.
-   * @param array|null $configuration
-   *   Optional configuration.
-   * @param int $flags
-   *   Set to STREAM_IS_URL if valid URL protocol. Default 0 (local stream).
-   *
-   * @return bool
-   *   True if the protocol was registered, false if not.
-   */
-  public static function register($protocol, FilesystemInterface $filesystem, ?array $configuration = NULL, $flags = 0) {
-    if (static::streamWrapperExists($protocol)) {
-      return FALSE;
-    }
-
-    static::$config[$protocol] = $configuration ?: static::$defaultConfiguration;
-    static::registerPlugins($protocol, $filesystem);
-    static::$filesystems[$protocol] = $filesystem;
-
-    return stream_wrapper_register($protocol, __CLASS__, $flags);
-  }
-
-  /**
-   * Unregisters a stream wrapper.
-   *
-   * @param string $protocol
-   *   The protocol.
-   *
-   * @return bool
-   *   True if the protocol was unregistered, false if not.
-   */
-  public static function unregister($protocol) {
-    if (!static::streamWrapperExists($protocol)) {
-      return FALSE;
-    }
-
-    unset(static::$filesystems[$protocol]);
-
-    return stream_wrapper_unregister($protocol);
-  }
-
-  /**
-   * Unregisters all controlled stream wrappers.
-   */
-  public static function unregisterAll() {
-    foreach (static::getRegisteredProtocols() as $protocol) {
-      static::unregister($protocol);
-    }
-  }
-
-  /**
-   * Returns array list of registered protocols.
-   *
-   * @return array
-   *   The list of registered protocols.
-   */
-  public static function getRegisteredProtocols() {
-    return array_keys(static::$filesystems);
-  }
-
-  /**
-   * Determines if a protocol is registered.
-   *
-   * @param string $protocol
-   *   The protocol to check.
-   *
-   * @return bool
-   *   True if it is registered, false if not.
-   */
-  protected static function streamWrapperExists($protocol) {
-    return in_array($protocol, stream_get_wrappers(), TRUE);
-  }
-
   /**
    * Registers plugins on the filesystem.
    *
@@ -1078,7 +1002,7 @@ class FlysystemStreamWrapper {
    *   True on success, false on failure.
    */
   private function touch($path) {
-    $path = $this->normalizeRelativePath($path);
+    $path = $this->normalizePath($path);
 
     $adapter = $this->filesystem->getAdapter();
 
@@ -1107,7 +1031,7 @@ class FlysystemStreamWrapper {
   public function rmdir($dirname, $options) {
     $this->uri = $dirname;
 
-    $dirname = $this->normalizeRelativePath($dirname);
+    $dirname = $this->normalizePath($dirname);
 
     if ($dirname === '') {
       throw new RootViolationException('Root directories can not be deleted.');
@@ -1147,7 +1071,7 @@ class FlysystemStreamWrapper {
    * @todo Refactor to make $dirname $uri instead.
    */
   public function mkdir($dirname, $mode, $options) {
-    $dirname = $this->normalizeRelativePath($dirname);
+    $dirname = $this->normalizePath($dirname);
 
     $adapter = $this->filesystem->getAdapter();
 
@@ -1180,8 +1104,8 @@ class FlysystemStreamWrapper {
    * @throws \Drupal\flysystem\StreamWrapper\Exception\NotADirectoryException
    */
   private function forcedRename($path, $newpath) {
-    $path = $this->normalizeRelativePath($path);
-    $newpath = $this->normalizeRelativePath($newpath);
+    $path = $this->normalizePath($path);
+    $newpath = $this->normalizePath($newpath);
 
     // Ignore useless renames.
     if ($path === $newpath) {
diff --git a/src/StreamWrapper/Plugin/Stat.php b/src/StreamWrapper/Plugin/Stat.php
index 4c20fe8..382d0cd 100644
--- a/src/StreamWrapper/Plugin/Stat.php
+++ b/src/StreamWrapper/Plugin/Stat.php
@@ -8,6 +8,8 @@ use Drupal\flysystemStreamWrapper\FlysystemStreamWrapper;
 
 /**
  * URL Stats utility.
+ *
+ * @codeCoverageIgnore
  */
 class Stat extends AbstractPlugin {
   /**
