diff --git a/src/Plugin/FlysystemUrlTrait.php b/src/Plugin/FlysystemUrlTrait.php
index ea682b2..66f05bd 100644
--- a/src/Plugin/FlysystemUrlTrait.php
+++ b/src/Plugin/FlysystemUrlTrait.php
@@ -31,12 +31,12 @@ trait FlysystemUrlTrait {
       'scheme' => $this->getScheme($uri),
       'filepath' => $path,
     ];
-
-    return Url::fromRoute('flysystem.serve', $arguments, ['absolute' => TRUE])
+    return Url::fromRoute('flysystem.serve', $arguments, ['absolute' => TRUE])->toString();
+    //return Url::fromRoute('flysystem.serve', $arguments, ['absolute' => TRUE])
       // Collect bubbleable metadata to not fall into the early rendering.
       /* @see \Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber::wrapControllerExecutionInRenderContext() */
-      ->toString(TRUE)
-      ->getGeneratedUrl();
+    //  ->toString(TRUE)
+    //  ->getGeneratedUrl();
   }
 
   /**
