diff --git a/phpunit.xml.dist b/phpunit.xml.dist
index 429dcf7..b3c6165 100644
--- a/phpunit.xml.dist
+++ b/phpunit.xml.dist
@@ -21,6 +21,8 @@
       <directory suffix=".module">./</directory>
       <directory suffix=".install">./</directory>
       <exclude>
+        <file>src/Asset/CssCollectionOptimizer.php</file>
+        <file>src/Asset/JsCollectionOptimizer.php</file>
         <directory suffix="Test.php">./</directory>
         <directory suffix="TestBase.php">./</directory>
         <directory suffix="bootstrap.php">./</directory>
